Source/WebCore/animation/ViewTimeline.cpp

```diff
@@ -365,6 +459,23 @@ const RenderBox* ViewTimeline::sourceScrollerRenderer() const
     return subjectRenderer->enclosingScrollableContainer();
 }
 
+const RenderElement* ViewTimeline::stickyContainer() const
+{
+    auto subject = m_subject.styleable();
+    if (!subject)
+        return nullptr;
+
+    CheckedPtr renderer = subject->renderer();
+
+    auto scrollerRenderer = sourceScrollerRenderer();
+    while (renderer && renderer != scrollerRenderer) {
+        if (renderer->isStickilyPositioned())
```

Comment:
What happens with nested stickies inside a scroller? Or a sticky inside a fixed position?

Reply:
Yeah both of these scenarios need some extra thought (and tests).